manim/mobject/geometry/line.py

```diff
@@ -125,16 +150,13 @@ def set_points_by_ends(
     init_points = generate_points
 
     def _account_for_buff(self, buff: float) -> None:
-        if buff == 0:
+        if buff <= 0:
             return
-        #
-        length = self.get_length() if self.path_arc == 0 else self.get_arc_length()
-        #
+        length = self.get_arc_length() if self.path_arc else self.get_length()
```

Comment:
Since `self.path_arc` is now always a `float`, the previous method with `if self.path_arc == 0` is a better option, because it's more explicit. Using `if variable` is best when `variable` is known to be a bool or possibly `None`.